src/wacks4shop/wack.py

```diff
@@ -21,25 +21,24 @@
 def main(db: Wackabase, dry=False):
     try:
         logger.info("TIME TO WACK")
-        logger.info("Dry run: %s", dry)
 
         discord = Discord(debug=dry)
         shift4shop = Shift4Shop(debug=dry)
 
-        now = datetime.now()
-
         last_timestamp = db.get_timestamp()
         sales = shift4shop.determine_sales(timestamp=last_timestamp)
-
         if not sales:
             return
 
         # grab the current number of total sales
         current_num_sales = shift4shop.get_num_sales()
         logger.info(f"current num sales: {current_num_sales}")
-        announce_new_sales(discord, sales, current_num_sales)
+        announce_new_sales(discord, sales, current_num_sales, id_type="shift4shop")
 
-        db.write_timestamp(now)
+        # write out the most recent sale's date
+        latest_sale = sales[-1].datetime
+        if latest_sale:
```

Comment:
is there any case where the last sale won't have a timestamp? is it better to not write a timestamp to db in this case or to use "now"? if we used "now" would we potentially drop sales?

Reply:
hm ok this was because I was lazy and was using the same `Sale` object between Etsy and Shift4Shop. Etsy sales we don't care about timestamp, while Shift4Shop we do. so I made a new Shift4Shop obj which will always have a datetime.

the trouble is `announce_new_sales` needs a `Sale` object, and python typing does not seem to understand that a `Shift4Shop` obj can be a `Sale` obj. https://github.com/allisonking/wacks-by-warby/pull/11/commits/16750fb698e96ff769885177a037e7e0acc35faa